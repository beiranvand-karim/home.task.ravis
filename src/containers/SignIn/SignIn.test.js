import {SignIn} from "./";
import {shallow} from "enzyme";
import React from 'react'
import renderer from 'react-test-renderer'
import {MemoryRouter} from "react-router-dom"

describe('SignIn Component', () => {
   const mockHistory = { push: jest.fn() };

   it('should sign in', () => {
      const wrapper = shallow(<SignIn history={mockHistory}/>);
      wrapper.find('[name="userName"]').simulate('change', {target: {name: 'userName', value: 'admin'}});
      wrapper.find('[name="passWord"]').simulate('change', {target: {name: 'passWord', value: 'admin'}});
      wrapper.find('form').simulate('submit', {preventDefault() {}});
      expect(mockHistory.push.mock.calls[0]).toEqual(['dashBoard'])
   });

   it('should matches the snapshot', () => {
      const tree = renderer.create(<MemoryRouter><SignIn/></MemoryRouter>);
      expect(tree).toMatchSnapshot()
   })
});
