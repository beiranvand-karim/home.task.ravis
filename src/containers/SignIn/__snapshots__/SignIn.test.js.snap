// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Component should matches the snapshot 1`] = `
<div
  className="signInContainer"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="inputElement"
      name="userName"
      onChange={[Function]}
      placeholder="userName"
      type="text"
      value=""
    />
    <input
      className="inputElement"
      name="passWord"
      onChange={[Function]}
      placeholder="passWord"
      type="password"
      value=""
    />
    <button
      type="submit"
    >
      sign in
    </button>
  </form>
</div>
`;
